optimism/src/lookups.rs

```diff
@@ -71,7 +72,54 @@ pub(crate) trait FixedLookupTables<F> {
     fn table_byte() -> LookupTable<F>;
 }
 
-impl<F: Field> FixedLookupTables<F> for LookupTable<F> {
+impl<F: Field> FixedLookupTables<F, LookupTableIDs> for LookupTable<F> {
+    fn in_table(id: LookupTableIDs, value: Vec<F>) -> Option<usize> {
+        let table = match id {
+            LookupTableIDs::RangeCheck16Lookup => Self::table_range_check_16().entries,
+            LookupTableIDs::SparseLookup => Self::table_sparse().entries,
+            LookupTableIDs::ResetLookup => Self::table_reset().entries,
+            LookupTableIDs::RoundConstantsLookup => Self::table_round_constants().entries,
+            LookupTableIDs::PadLookup => Self::table_pad().entries,
+            LookupTableIDs::ByteLookup => Self::table_byte().entries,
+            _ => return None,
+        };
+        let idx = value[0]
```

Comment:
There is an assumption that the index is the first element (which is true). But it might be worth to keep this invariant somewhere in a type or make it clear somewhere.

Reply:
Yeah, this holds for these tables in particular, but it could be untrue for others. I will add one comment. Another "more general" option would be to just look inside the whole table, but that seemed too inefficient considering the table lengths if one can know where exactly each entry should sit on.